### PR TITLE
EngineJobHashingActor tests (feat. CachingConfigSpec redux)

### PR DESCRIPTION
**Pull request by cjllanwarne**

### CODE REVIEW

**Comment by kshakir** [comment]
LGTM 👍

File: engine/src/main/scala/cromwell/engine/workflow/lifecycle/MaterializeWorkflowDescriptorActor.scala [hunk 1]
```diff
@@ -72,6 +72,36 @@ object MaterializeWorkflowDescriptorActor {
   case class MaterializeWorkflowDescriptorActorData()
 
   private val DefaultWorkflowFailureMode = NoNewCalls.toString
+
+  private[lifecycle] def validateCallCachingMode(workflowOptions: WorkflowOptions, conf: Config): ErrorOr[CallCachingMode] = {
+
+    def readOptionalOption(option: WorkflowOption): ErrorOr[Boolean] = {
+      workflowOptions.getBoolean(option.name) match {
+        case Success(x) => x.successNel
```
Comment (mcovarr) [inline engine/src/main/scala/cromwell/engine/workflow/lifecycle/MaterializeWorkflowDescriptorActor.scala]: TIL these default to `true`.

File: engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/EngineJobExecutionActor.scala [hunk 2]
```diff
@@ -190,7 +190,7 @@ class EngineJobExecutionActor(jobKey: BackendJobDescriptorKey,
   }
 
   def initializeJobHashing(jobDescriptor: BackendJobDescriptor, activity: CallCachingActivity) = {
-    val props = EngineJobHashingActor.props(jobDescriptor, initializationData, factory.fileContentsHasherActor,
+    val props = EngineJobHashingActor.props(self, jobDescriptor, initializationData, factory.fileContentsHasherActor,
       callCacheReadActor, dockerHashLookupActor, factory.runtimeAttributeDefinitions(initializationData), backendName, activity)
```
Comment (mcovarr) [inline engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/EngineJobExecutionActor.scala]: looks like an ideal application for keyword arguments

File: engine/src/test/scala/cromwell/engine/workflow/lifecycle/execution/callcaching/EngineJobHashingActorSpec.scala [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+    val callCacheReadActor = system.actorOf(Props(new PredictableCallCacheReadActor(cacheLookupResponses)))
+    system.actorOf(EngineJobHashingActor.props(replyTo, jobDescriptor, initializationData, fileHashingActor.getOrElse(emptyActor), callCacheReadActor, dockerHashLookupActor.getOrElse(emptyActor), runtimeAttributeDefinitions, backendName, activity))
+  }
+
+  def emptyActor(implicit actorSystem: ActorSystem) = actorSystem.actorOf(Props.empty)
+
+  def templateJobDescriptor
+  (
+    inputs: Map[LocallyQualifiedName, WdlValue] = Map.empty
+  ) = {
```
Comment (mcovarr) [inline engine/src/test/scala/cromwell/engine/workflow/lifecycle/execution/callcaching/EngineJobHashingActorSpec.scala]: some artistic parameter lists here

**Comment by mcovarr** [comment]
:+1:

**Pull request merged**